test/integration/targets/systemd/tasks/main.yml

```diff
@@ -79,4 +79,43 @@
       environment:
           SYSTEMD_OFFLINE: 1
 
+    - name: Create dummy service
+      copy:
+        src: dummy.service
+        dest: /etc/systemd/system/dummy.service
+        owner: root
+        mode: 0640
+
+    - name: try-restart result should not restart dummy service when it is not running
+      systemd:
+        name: dummy
+        state: tried-restarting
+      register: inactive_try_restart
+
+    - name: Start dummy service
+      systemd:
+        name: dummy
+        state: started
+
+    - name: try-restart result should restart dummy serivce when it is running
+      systemd:
+        name: dummy
+        state: tried-restarting
+      register: running_try_restart
+
+    - assert:
+        that:
+          - 'running_try_restart is changed'
+          - 'inactive_try_restart is not changed'
+
+    - name: Stop dummy service
+      systemd:
+        name: dummy
+        state: stopped
+
+    - name: Remove dummy service
+      file:
+        path: /etc/systemd/system/dummy.service
+        state: absent
```

Comment:
These tasks should be handlers that are notified by the `copy` task above. That way this is guaranteed to be removed even in the event of a failure since we run Ansible with `DEFAULT_FORCE_HANDLERS=True` in our test environment.

Reply:
Also add a handler task to reload systemd once the unit file has been removed. Probably not necessary, but better safe than sorry since the test instances are shared.

```yaml
    - name: Reload systemd
      systemd:
        daemon-reload: yes
```

Reply:
Fixed in https://github.com/ansible/ansible/pull/71172/commits/c57ef436b62bddba159e3ab811663388b2137977 and https://github.com/ansible/ansible/pull/71172/commits/e97b8e32482c6a27c17e27a47cc20192dfb284f3.